### PR TITLE
IG Checklist Fixes + Extra Resource Support

### PR DESCRIPTION
**Pull request by cmoesel**
This PR adds two main features:
1. Better support for the new requirements asked of HL7 IGs (see checklist image below)
2. Better support for including `SearchParameter`, `OperationDefinition`, `CapabilityStatement`, and `Conformance` resources in the exported IG (even though CIMPL can't produce them itself).

To test this, switch all branches to `5.x`, and then to `ig-checklist-fixes`.  When done, your branches should look something like this:

```
shr-adl-bmm-export       [master]
shr-cli                  [ig-checklist-fixes]
shr-data-dict-export     [5.x]
shr-es6-export           [5.x]
shr-expand               [5.x]
shr-fhir-export          [ig-checklist-fixes]
shr-grammar              [5.x]
shr-json-export          [master]
shr-json-javadoc         [5.x]
shr-json-schema-export   [5.x]
shr-models               [5.x]
shr-test-helpers         [5.x]
shr-text-import          [5.x]
```

Then clone my [us-drug-formularies](https://gitlab.mitre.org/cmoesel/us-drug-formularies) fork and switch to `ig-checklist-fixes` branch.  Now run the CLI against it and then run it through the publisher (NOTE: This requires you to be on a network *without* a proxy):

```sh
$ node . -c ig-usdrugformulary-config.json /path/to/us-drug-formularies/
```

```sh
$ yarn ig:publish
```

Now open the IG (`yarn ig:open`) and validate that it passes the following checklist:
image

(Note, it is ok that the yellow box contains placeholder text).

Also check that the menu has an option for `Search Parameters` and the corresponding page links to two search parameters.

If you want to test the other resource types, you can download some from US Core R4 and put them in the `us-drug-formularies` `searchParameters_usdrugformulary` folder (don't worry about the misleading name):
* [OperationDefinition-docref.json](https://build.fhir.org/ig/HL7/US-Core-R4/OperationDefinition-docref.json)
* [CapabilityStatement-us-core-server.json](https://build.fhir.org/ig/HL7/US-Core-R4/CapabilityStatement-us-core-server.json)
* [CapabilityStatement-us-core-client.json](https://build.fhir.org/ig/HL7/US-Core-R4/CapabilityStatement-us-core-client.json)

Then run the CLI and IG Publisher again.  The menu should have items for the new resources.

### CODE REVIEW

**Comment by cmoesel** [comment]
Please review only (do not merge) -- I forgot to mark it as DRAFT.  Companion PR at: https://github.com/standardhealth/shr-cli/pull/215

**Comment by kjmahalingam** [comment]
The code of this and the associated PR looks good! I do have a few questions though. Pictures of what I see attached. 

USDFHeader

USDFFooter

I believe I can tick all the check boxes, but am unsure about a couple in the footer. Notably, 'Propose Changes', 'History Link', and 'Copyright Year' seem to be missing. That is, of course, unless I'm missing something.

I believe I followed the instructions you provided correctly. Anything I could be missing?

**Comment by cmoesel** [comment]
@kjmahalingam -- Thank you.  Are you sure you were running against my `ig-checklist-fixes` branch of the `us-drug-formularies` CIMPL project?  That's the only reason I can think those wouldn't show up.

**Comment by kjmahalingam** [comment]
Ah, you know what, @cmoesel , that might be the issue. I'm not sure I did that part. Let me try again.

**Comment by kjmahalingam** [comment]
This fixed a few issues but I still see some. Image of footer attached.

USDFFooter2

The 'Propose Changes' link works and the history link exists. However, when I click the history link, it gives a 404 at this site: `http://hl7.org/fhir/us/Davinci-drug-formulary/history.html`. Additionally, I still don't see a copyright year. Any thoughts?

**Comment by cmoesel** [comment]
I should have mentioned: we _expect_ the history link to 404.  HL7 hasn't published the history page yet.

As for copyright year, I'll look into that.  Maybe I messed something up.

**Comment by cmoesel** [comment]
OK.  Found the issue w/ copyright year and pushed up a fix.

**Review by kjmahalingam — APPROVED** [review]
All good!

**Comment by cmoesel** [comment]
FYI - to support easier cherry-picking to 6.x, I've rebased the commits into two commits: a commit w/ all code changes, and a separate commit w/ version update.
